@here/olp-sdk-dataservice-read/test/unit/CatalogClient.test.ts

```diff
@@ -132,6 +134,32 @@ describe("CatalogClient", () => {
         assert.isTrue(response.versions.length > 0);
     });
 
+    it("Should method getEarliestVersion provide the minimun version availiable for the given catalogRequest", async () => {
+        const testBillingTag = "testBillingTag";
+        const mockedEarliestVersion: MetadataApi.VersionResponse = {
+            version: 5
+        };
+
+        getEarliestVersionsStub.callsFake(
+            (
+                builder: any,
+                params: any
+            ): Promise<MetadataApi.VersionResponse> => {
+                return Promise.resolve(mockedEarliestVersion);
+            }
+        );
+
+        const catalogRequest = new dataServiceRead.CatalogVersionRequest().withBillingTag(
```

Comment:
Please do not use the real class in the unit test, just create mocked or stubbed instance of CatalogVersionRequest and use it

Reply:
Updated